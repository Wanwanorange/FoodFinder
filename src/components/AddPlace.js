import React, { Component } from 'react';
import PlacesAutocomplete from 'react-places-autocomplete';
import { addPlace } from "../actions/place-actions";
import { connect } from 'react-redux';
import selectPlaces from '../selectors/place-selector';
import '../styles/AddPlace.css';

export class AddPlace extends Component {
    state = {
        id: '',
        name: '',
        category: 'bakery',
        completed: false,
        error: ''
    };

    addPlaceIfNoErrors = (input) => {
        const name = input.name;
        if (!name) {
            return 'Please enter place name';
        }
        else if (this.props.places.findIndex(place => place.name === name) > -1) {
            return 'This place is already on your list';
        }
        this.props.addPlace({
            id: input.id,
            name,
            category: input.category,
            completed: input.completed
        });
        return '';
    };

    onSubmit = (e) => {
        e.preventDefault();
        const error = this.addPlaceIfNoErrors(this.state);

        this.setState({ error });

        if (!error) {
            document.getElementsByName('name').value = '';
        }
    };

    onNameChange = (e) => {
        const name = e;
        this.setState(() => ({ id: name, name }));
    };

    onCategoryChange = (e) => {
        const category = e.target.value;
        this.setState(() => ({ category }));
    };

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit} >
                    <PlacesAutocomplete
                        value={this.state.name}
                        onChange={this.onNameChange}>
                        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
                            <div>
                                <input
                                    {...getInputProps({
                                        placeholder: 'Search Places ...',
                                        className: 'location-search-input',
                                    })}
                                />
                                {suggestions.length > 0 && (
                                    <div className="autocomplete-dropdown-container">
                                        {loading && <div>Loading...</div>}
                                        {suggestions.map(suggestion => {
                                            const className = suggestion.active
                                                ? 'suggestion-item--active'
                                                : 'suggestion-item';
                                            // inline style for demonstration purpose
                                            const style = suggestion.active
                                                ? { backgroundColor: '#fafafa', cursor: 'pointer' }
                                                : { backgroundColor: '#ffffff', cursor: 'pointer' };
                                            return (
                                                <div
                                                    {...getSuggestionItemProps(suggestion, {
                                                        className,
                                                        style,
                                                    })}
                                                >
                                                    <span>{suggestion.description}</span>
                                                </div>
                                            );
                                        })}
                                    </div>
                                )}
                            </div>
                        )}
                    </PlacesAutocomplete>

                    <select name="category"
                        onChange={this.onCategoryChange}>
                        <option value="bakery">Bakery</option>
                        <option value="cafe">Cafe</option>
                        <option value="meal delivery">Meal delivery</option>
                        <option value="meal takeaway">Meal Takeaway</option>
                        <option value="restaurant">Restaurant</option>
                    </select>
                    <button>Add Place</button>
                </form>
                {this.state.error && <p className="error">{this.state.error}</p>}
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        places: selectPlaces(state.places, false)
    };
};

const mapDispatchToProps = (dispatch) => ({
    addPlace: (place) => dispatch(addPlace(place))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddPlace);
