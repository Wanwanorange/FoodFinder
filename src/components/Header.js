import React from 'react';
import Select from 'react-select';
import { NavLink } from 'react-router-dom';
import '../styles/Header.css';

const filterOps = [
    { value: 'bakery', label: 'Bakery' },
    { value: 'cafe', label: 'Cafe' },
    { value: 'meal delivery', label: 'Meal Delivery' },
    { value: 'meal takeway', label: 'Meal Takeaway' },
    { value: 'restaurant', label: 'Restaurant' },
];

const Header = () => (
    <div>
        <header>
            <div className="tabs">
                <NavLink className="tab" activeClassName="is-active" to="/" exact={true}>Places to Go</NavLink>
                <NavLink className="tab" activeClassName="is-active" to="/completed">Places I've Been</NavLink>
            </div>
            <Select
                    className="filter"
                    isMulti
                    options={filterOps}
                    placeholder="Filter categories..."/>
        </header>
    </div>
);

export default Header;