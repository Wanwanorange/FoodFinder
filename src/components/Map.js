import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';
import { selectAll } from '../selectors/place-selector';
import { connect } from 'react-redux';
import marker from '../img/marker.png';
import completedmarker from '../img/completedmarker.png';

const AnyReactComponent = ({ completed }) => (
  <div>{completed ? <img src={completedmarker}/> : <img src={marker}/>}</div>
);

class SimpleMap extends Component {
  static defaultProps = {
    center: {
      lat: 44.5,
      lng: -85.5
    },
    zoom: 5
  };

  render() {
    return (
      <div style={{ height: '60vh', width: '100%' }}>
        <GoogleMapReact
          bootstrapURLKeys={{ key: 'AIzaSyDVpveOb7pwNDm9fll34qPkpWFdje4lxkw' }}
          defaultCenter={this.props.center}
          defaultZoom={this.props.zoom}
        >
          {this.props.places.map((place) =>
            <AnyReactComponent 
            lat={place.latitude} 
            lng={place.longitude} 
            text={place.name}
            completed={place.completed}
            />)}
        </GoogleMapReact>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    places: selectAll(state.places)
  };
};

export default connect(mapStateToProps)(SimpleMap);