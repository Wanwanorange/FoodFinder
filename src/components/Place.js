import React, { Component } from 'react';
import '../styles/Place.css';

class Place extends Component {

    removePlace = () => {
        this.props.removePlace({ id: this.props.id });
    }

    completePlace = () => {
        this.props.completePlace({ id: this.props.id });
    }

    render() {
        return (
            <div className="place">
                <div className="placeCheck">
                    <p>{this.props.name}, {this.props.category}</p>
                    {!this.props.completed &&
                        <button className="completeButton"
                            onClick={this.completePlace}>Been there!</button>}
                </div>
                <button className="removeButton"
                    onClick={this.removePlace} id={this.props.id}>X</button>
            </div>
        );
    }
}

export default Place