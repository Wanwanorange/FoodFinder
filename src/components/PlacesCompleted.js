import React, {Component} from 'react';
import Place from "./Place";

class PlacesCompleted extends Component {
    constructor(props) {
        super(props);
        this.state = {
            completedPlaces: [<Place name="Jenis Ice Cream" category="cafe"/>]
        };
    }

    render() {
        return (
            <div>
                <ul>
                    {this.state.completedPlaces}
                </ul>
            </div>
        );
    }
}

export default PlacesCompleted;