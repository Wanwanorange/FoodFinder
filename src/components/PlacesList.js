import React from 'react';
import Place from './Place';
import { removePlace, completePlace } from "../actions/place-actions";
import selectPlaces from '../selectors/place-selector';

import { connect } from 'react-redux';

export const PlacesList = (props) => (
    <div>
        {props.places.map((place) =>
            <Place {...place}
                removePlace={props.removePlace}
                completePlace={props.completePlace} />)}
    </div>
);

const mapStateToProps = (state) => {
    return {
        places: selectPlaces(state.places, false)
    };
};

const mapDispatchToProps = (dispatch) => ({
    removePlace: (data) => dispatch(removePlace(data)),
    completePlace: (data) => dispatch(completePlace(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(PlacesList);