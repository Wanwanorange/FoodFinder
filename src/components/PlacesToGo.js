import React from 'react';
import '../styles/PlacesToGo.css';
import AddPlace from './AddPlace';
import PlacesList from './PlacesList';

const PlacesToGo = () => (
    <div className="PlacesToGo">
        <AddPlace/>
        <PlacesList/>
    </div>
);


export default PlacesToGo;
