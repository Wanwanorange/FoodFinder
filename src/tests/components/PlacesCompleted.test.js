import React from 'react';
import { shallow } from 'enzyme';
import { PlacesCompleted } from '../../components/PlacesCompleted';
import places from '../testData/places';

test('should render PlacesCompleted correctly', () => {
    const wrapper = shallow(<PlacesCompleted places={places}/>);
    expect(wrapper).toMatchSnapshot();
});