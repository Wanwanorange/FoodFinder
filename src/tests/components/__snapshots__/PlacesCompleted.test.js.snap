// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlacesCompleted correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    places={
        Array [
            Object {
              "category": "meal delivery",
              "completed": false,
              "id": "1234",
              "name": "Big Mamas",
            },
            Object {
              "category": "cafe",
              "completed": true,
              "id": "2903",
              "name": "Kanga Wanga",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Place
          category="meal delivery"
          completed={false}
          id="1234"
          name="Big Mamas"
          removePlace={undefined}
/>,
        <Place
          category="cafe"
          completed={true}
          id="2903"
          name="Kanga Wanga"
          removePlace={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "category": "meal delivery",
          "completed": false,
          "id": "1234",
          "name": "Big Mamas",
          "removePlace": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "category": "cafe",
          "completed": true,
          "id": "2903",
          "name": "Kanga Wanga",
          "removePlace": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Place
            category="meal delivery"
            completed={false}
            id="1234"
            name="Big Mamas"
            removePlace={undefined}
/>,
          <Place
            category="cafe"
            completed={true}
            id="2903"
            name="Kanga Wanga"
            removePlace={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "category": "meal delivery",
            "completed": false,
            "id": "1234",
            "name": "Big Mamas",
            "removePlace": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "category": "cafe",
            "completed": true,
            "id": "2903",
            "name": "Kanga Wanga",
            "removePlace": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
